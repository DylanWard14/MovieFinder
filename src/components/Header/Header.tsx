import React from 'react';
import styled from 'styled-components/macro';

import User from '../User/User';
import { UserDetailsState } from '../../store/User-details/types';
import { UserAuthState } from '../../store/User-auth/types';

const Component = styled.header`
    height: 2rem;
    color: white;
    padding: 0.5rem 1rem;
    display: flex;
    justify-content: space-between;
    align-items: center;
`;

const Heading = styled.h1`
    margin: 0;
    background: linear-gradient(45deg, ${props => props.theme.colors.main}, ${props => props.theme.colors.blue});
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    font-size: 1.5rem;
    user-select: none;

    &::after,
    &::before {
        width: 100%;
        height: 0.25rem;
        background: linear-gradient(45deg, ${props => props.theme.colors.main}, ${props => props.theme.colors.blue});
        content: '';
        display: block;
        border-radius: 5px;
    }
`;

export interface HeaderProps {
    userAuth: UserAuthState;
    userDetails: UserDetailsState;
    login: () => void;
}

const Header: React.FunctionComponent<HeaderProps> = ({ userAuth, userDetails, login }) => {
    return (
        <Component>
            <div>
                <Heading>Movie Finder</Heading>
            </div>
            <div>
                <User userAuth={userAuth} userDetails={userDetails} handleLogin={login} />
            </div>
        </Component>
    );
};

export default Header;
