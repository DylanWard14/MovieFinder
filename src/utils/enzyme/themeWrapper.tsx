import React from 'react';
import { render, shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import myTheme from '../../theme';

export const renderWrapper = (component: React.ReactNode) => {
    return render(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const shallowWrapper = (component: React.ReactNode) => {
    return shallow(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};

export const mountWrapper = (component: React.ReactNode) => {
    return mount(<ThemeProvider theme={myTheme}>{component}</ThemeProvider>);
};
