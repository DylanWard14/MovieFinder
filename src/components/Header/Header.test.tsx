import React from 'react';
import { renderWrapper } from '../../utils/enzyme/themeWrapper';

import Header from './Header';

describe('Header', () => {
    const login = jest.fn();
    const setup = (propsOverrides = {}) => {
        const props = {
            userAuth: {
                loading: false,
                data: undefined,
            },
            userDetails: {
                loading: false,
                data: undefined,
            },
            login: login,
            ...propsOverrides,
        };

        return {
            rendered: renderWrapper(<Header {...props} />),
        };
    };
    describe('snapshot', () => {
        it('renders correctly', () => {
            const { rendered } = setup();

            expect(rendered).toMatchSnapshot();
        });
    });
});
