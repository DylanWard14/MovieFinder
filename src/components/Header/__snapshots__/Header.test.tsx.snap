// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot renders correctly 1`] = `
<header
  class="Header__Component-sc-3q01gu-0 hoCHew"
>
  <div>
    <h1
      class="Header__Heading-sc-3q01gu-1 jPDuAc"
    >
      Movie Finder
    </h1>
  </div>
  <div>
    <button
      class="Button__StyledButton-tmhdcf-0 kGMZFd"
    >
      Login
    </button>
  </div>
</header>
`;
